deps/chakrashim/src/v8debug.cc

```diff
@@ -21,71 +21,715 @@
+    messageQueue.SaveMessage(command, length);
+
+    // Request async break from engine so that we can process
+    // commands on JsDiagDebugEventAsyncBreak
+    JsErrorCode errorCode = JsDiagRequestAsyncBreak(
+      jsrt::IsolateShim::FromIsolate(isolate)->GetRuntimeHandle());
+    CHAKRA_VERIFY_NOERROR(errorCode);
   }
 
-error:
-#else
-  hr = E_FAIL;  // ChakraCore does not support JsStartDebugging
+  static JsValueRef CALLBACK Log(
+    JsValueRef callee,
+    bool isConstructCall,
+    JsValueRef *arguments,
+    unsigned short argumentCount,
+    void *callbackState) {
+    JsValueRef scriptRef;
+    const wchar_t *script;
+
+    if (argumentCount > 1 &&
+      jsrt::ToString(arguments[1], &scriptRef, &script) == JsNoError) {
+      fwprintf(stdout, L"%s\n", script);
+#ifdef DEBUG
+      flushall();
 #endif
+    }
+
+    return JS_INVALID_REFERENCE;
+  }
+
+  static JsValueRef CALLBACK SendDelayedRespose(
+    JsValueRef callee,
+    bool isConstructCall,
+    JsValueRef *arguments,
+    unsigned short argumentCount,
+    void *callbackState) {
+    JsValueType resultType;
+    JsErrorCode errorCode = JsGetValueType(arguments[1], &resultType);
+    CHAKRA_VERIFY_NOERROR(errorCode);
+
+    if (resultType == JsString) {
+      v8::Local<v8::String> v8Str = MessageImpl::JsValueRefToV8String(
+        arguments[1]);
+      v8::MessageImpl* msgImpl = new v8::MessageImpl(v8Str);
+      Debug::handler(*msgImpl);
+    }
+
+    return JS_INVALID_REFERENCE;
+  }
+
+  static JsValueRef CALLBACK JsGetScripts(
+    JsValueRef callee,
+    bool isConstructCall,
+    JsValueRef *arguments,
+    unsigned short argumentCount,
+    void *callbackState) {
+    JsValueRef sourcesList;
+    JsErrorCode errorCode = JsDiagGetScripts(&sourcesList);
+    CHAKRA_VERIFY_NOERROR(errorCode);
+
+    return sourcesList;
+  }
+
+  static JsValueRef CALLBACK JsGetSource(
+    JsValueRef callee,
+    bool isConstructCall,
+    JsValueRef *arguments,
+    unsigned short argumentCount,
+    void *callbackState) {
+    int scriptId;
+    JsValueRef source = JS_INVALID_REFERENCE;
+    if (argumentCount > 1 &&
+      jsrt::ValueToInt(arguments[1], &scriptId) == JsNoError) {
+      JsErrorCode errorCode = JsDiagGetSource(scriptId, &source);
+      CHAKRA_VERIFY_NOERROR(errorCode);
+    }
+
+    return source;
+  }
+
+  static JsValueRef CALLBACK JsSetStepType(
+    JsValueRef callee,
+    bool isConstructCall,
+    JsValueRef *arguments,
+    unsigned short argumentCount,
+    void *callbackState) {
+    bool success = false;
+    int stepType;
+    if (argumentCount > 1 &&
+      jsrt::ValueToInt(arguments[1], &stepType) == JsNoError) {
+      JsErrorCode errorCode = JsDiagSetStepType((JsDiagStepType)stepType);
+      CHAKRA_VERIFY_NOERROR(errorCode);
+      success = true;
+    }
+
+    JsValueRef returnRef = JS_INVALID_REFERENCE;
+    JsBoolToBoolean(success, &returnRef);
+
+    return returnRef;
+  }
+
+  static JsValueRef CALLBACK JsSetBreakpoint(
```

Comment:
>JsSetBreakpoint [](start = 29, length = 15)

Can you different nomenclature for these methods? Usually Js*() methods are Jsrt APIs. Would be good to reflect that these are shim methods.